tests/unittests/eventengine/severity_test.py

```diff
@@ -0,0 +1,245 @@
+    ):
+        rules = SeverityRules.load(simple_severity_ruleset_source)
+        alert = Mock()
+        alert.foo = "perfectly-cromulent"
+        assert rules.evaluate(alert) == 3
+
+    def test_should_add_to_severity_for_nested_rule(
+        self, nested_severity_ruleset_source
+    ):
+        rules = SeverityRules.load(nested_severity_ruleset_source)
+        alert = Mock()
+        alert.first = "level-one"
+        alert.second = "level-two"
+        assert rules.evaluate(alert) == 4
+
+    def test_should_properly_set_severity_based_on_nested_attributes(
+        self, nested_severity_ruleset_source
+    ):
+        rules = SeverityRules.load(nested_severity_ruleset_source)
+        alert = Mock()
+        alert.netbox.room.id = "Milliways"
+        alert.netbox.room.location.id = "The end of the Universe"
+        alert.netbox.category = "SW"
+        assert rules.evaluate(alert) == 1
+
+    def test_should_reset_severity_according_to_rule(
+        self, nested_severity_ruleset_source
+    ):
+        rules = SeverityRules.load(nested_severity_ruleset_source)
+        alert = Mock()
+        alert.foo = "0xc0ffee"
+        alert.bar = "0xbadc0de"
+        assert rules.evaluate(alert) == 5
+
+    def test_should_evaluate_to_single_rule_on_empty_rulesets(
+        self, empty_ruleset_source
+    ):
+        """Tests that an empty ruleset is parsed into a single default-value rule """
+        rules = SeverityRules.load(empty_ruleset_source)
+        assert len(rules) == 1
+
+    def test_should_return_original_severity_when_no_default_is_set(
+        self, empty_ruleset_source
+    ):
+        rules = SeverityRules.load(empty_ruleset_source)
+        alert = Mock()
+        alert.severity = 1
+        assert rules.evaluate(alert) == 1
+
+    def test_should_ignore_broken_rules(self, broken_ruleset_source):
+        rules = SeverityRules.load(broken_ruleset_source)
+
+        class MockAlert:
+            """Simple Alert mock that does NOT ignore illegal attribute accesses"""
+
+            severity = 5
+
+        assert rules.evaluate(MockAlert()) == 5
+
+
+#
+# Fixtures
+#
+@pytest.fixture(scope="session")
+def simple_severity_ruleset(simple_severity_ruleset_source):
+    return yaml.safe_load(simple_severity_ruleset_source)
+
+
+@pytest.fixture(scope="session")
+def simple_severity_ruleset_source():
+    return """
+---
+default-severity: 2
+rules:
+  - foo: perfectly-cromulent
+    severity: '+1'
+"""
+
+
+@pytest.fixture(scope="session")
+def nested_severity_ruleset(nested_severity_ruleset_source):
+    return yaml.safe_load(nested_severity_ruleset_source)
+
+
+@pytest.fixture(scope="session")
+def nested_severity_ruleset_source():
+    return """
+---
+default-severity: 2
+rules:
+  - first: level-one
+    severity: '+1'
+    rules:
+    - second: level-two
+      severity: '+1'
+  - foo: '0xc0ffee'
+    bar: '0xbadc0de'
+    severity: 5
+  - netbox.category: SW
+    netbox.room.id: Milliways
+    severity: '-1'
```

Comment:
Negative modifiers are not documented but obviously they are supposed to work, according to this test. How about modfiers higher than `+1` and less than `-1`? Explicit is very much better than implicit when it comes to documenting a DSL...

Reply:
I see from the grammar parser (I read that one last since that takes more time) that both + and - and any integer is ok for value modifiers. Still need to be in the user docs, and preferrably also in the tests.

Reply:
Agreed that the docs can be more explicit about this.

Not a fan of random numbers in tests, though - but I can pick another number than 1, just for show.

Reply:
I changed one piece of test data (not this one, since -1 will take the test alert down to 1, using -2 would also just take it down to 1, so the test wouldn't change even if the fixture did)